src/github.com/getlantern/autoupdate/autoupdate.go

```diff
@@ -0,0 +1,169 @@
+// Package autoupdate provides Lantern with special tools to autoupdate itself
+// with minimal effort.
+package autoupdate
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/getlantern/flashlight/util"
+	"github.com/getlantern/go-update"
+	"github.com/getlantern/go-update/check"
+)
+
+const noVersion = -1
+
+// Making sure AutoUpdate and Patch satisfy AutoUpdater and Patcher.
+var (
+	_ = AutoUpdater(&AutoUpdate{})
+	_ = Patcher(&Patch{})
+)
+
+var (
+	// How much time should we wait between update attempts?
+	sleepTime = time.Hour * 4
+)
+
+// SetProxy sets the proxy to use.
+func SetProxy(proxyAddr string) {
+	var err error
+
+	if proxyAddr != "" {
+		// Create a HTTP proxy and pass it to the update package.
+		if update.HTTPClient, err = util.HTTPClient("", proxyAddr); err != nil {
+			log.Printf("Could not use proxy: %q\n", err)
+		}
+	} else {
+		update.HTTPClient = &http.Client{}
+	}
+}
+
+// AutoUpdate satisfies AutoUpdater and can be used for other programs to
+// configure automatic updates.
+type AutoUpdate struct {
+	cfg config
+	v   int
+	// When a patch has been applied, the patch's version will be sent to
+	// UpdatedTo.
+	UpdatedTo chan int
+}
+
+// New creates an AutoUpdate struct based in the configuration defined in
+// config.go.
+func New(appName string) *AutoUpdate {
+	if configMap[appName] == nil {
```

Comment:
Yeah, I think the need to decouple from equinox at the API level isn't really that important, since there are steps that the user has to take outside of using autoupdate in order to even publish updates, which are all very equinox specific.  I would just accept the configMap data structure as is for now.